File: solutions/python/rectangles/1/rectangles.py
def rectangles(strings):
    if not strings:
        return 0
    rows, cols = len(strings), len(strings[0])
    corners = [(r, c) for r in range(rows) for c in range(cols) if strings[r][c] == '+']
    count = 0
    for i, (r1, c1) in enumerate(corners):
        for r2, c2 in corners[i+1:]:
            if r2 > r1 and c2 > c1:
                if all(strings[r1][c] in '-+' for c in range(c1, c2+1)) and \
                   all(strings[r2][c] in '-+' for c in range(c1, c2+1)) and \
                   all(strings[r][c1] in '|+' for r in range(r1, r2+1)) and \
                   all(strings[r][c2] in '|+' for r in range(r1, r2+1)):
                    count += 1
    return count

File: solutions/javascript/rectangles/1/rectangles.js
export function count(rows) {
  let result = 0;
  const corners = [];
  for (let r = 0; r < rows.length; r++) {
    for (let c = 0; c < rows[r].length; c++) {
      if (rows[r][c] === '+') {
        corners.push([r, c]);
      }
    }
  }
  for (let i = 0; i < corners.length; i++) {
    for (let j = i + 1; j < corners.length; j++) {
      const [r1, c1] = corners[i];
      const [r2, c2] = corners[j];
      if (r1 < r2 && c1 < c2) {
        if (rows[r1][c2] !== '+' || rows[r2][c1] !== '+') continue;
        let valid = true;
        for (let c = c1 + 1; c < c2; c++) {
          if (rows[r1][c] !== '-' && rows[r1][c] !== '+') {
            valid = false;
            break;
          }
        }
        if (!valid) continue;
        for (let c = c1 + 1; c < c2; c++) {
          if (rows[r2][c] !== '-' && rows[r2][c] !== '+') {
            valid = false;
            break;
          }
        }
        if (!valid) continue;
        for (let r = r1 + 1; r < r2; r++) {
          if (rows[r][c1] !== '|' && rows[r][c1] !== '+') {
            valid = false;
            break;
          }
        }
        if (!valid) continue;
        for (let r = r1 + 1; r < r2; r++) {
          if (rows[r][c2] !== '|' && rows[r][c2] !== '+') {
            valid = false;
            break;
          }
        }
        if (valid) result++;
      }
    }
  }
  return result;
}
